### ISSUE TITLE
Training convergence problem

### ISSUE THREAD

**Issue by alala521** [KuanchihHuang/Reason3D#14]
![Image](https://github.com/user-attachments/assets/88381cc0-cd20-400e-a236-faade35ce06b)
The first stage of non convergence in the search branch was trained on 6 3090s with a batch size of 8

**Comment by KuanchihHuang** [comment]
Hi, our setup uses four GPUs with batch size of 16. If you choose a different batch size, please adjust the learning rate accordingly.

**Comment by Catchyl-22** [comment]
> Hi, our setup uses four GPUs with batch size of 16. If you choose a different batch size, please adjust the learning rate accordingly.

Hi,i set my yaml like this:

  lr_sched: "linear_warmup_cosine_lr"
  init_lr: 1e-3
  min_lr: 1e-4
  warmup_lr: 1e-7

  warmup_steps: 3000

  weight_decay: 0.05
  max_epoch: 100
  batch_size_train: 64
  batch_size_eval: 1
  num_workers: 4
  accum_grad_iters: 1


but it's result turn out to be like this:


===================================

3D Reasoning segmentation on ScanNet (308 samples):
Val result: mIoU/Acc50/Acc25 0.2009/0.1948/0.2857

===================================

3D Reasoning segmentation on Matterport3D (837 samples):
Val result: mIoU/Acc50/Acc25 0.1938/0.1732/0.3226

it's a little far from the result in paper. i wonder how many epochs have you trained to get the result in the paper

**Comment by KuanchihHuang** [comment]
Hi, we trained our model for 100 epochs.
Did you specify the pretrained weights of the referring segmentation model when training the reasoning segmentation model?

**Comment by Catchyl-22** [comment]
> Hi, we trained our model for 100 epochs. Did you specify the pretrained weights of the referring segmentation model when training the reasoning segmentation model?
Thansk for reply i use the way in reademe and the referf resulet is : Val result: mIoU/Acc50/Acc25 0.4059/0.4271/0.5397  it's almost the same as the paper but the reason3d part is a little worse . i think maybe it;s because the searh branch just trained on mattport

**Comment by zhao-py** [comment]
> ![Image](https://github.com/user-attachments/assets/88381cc0-cd20-400e-a236-faade35ce06b) The first stage of non convergence in the search branch was trained on 6 3090s with a batch size of 8



Sorry to bother you. I wonder if you can run the code normally. I met a similar question in #15; the model doesn't converge at all. Can you give me some advice? Looking forward to your reply.
